builtInFunctions/saveUserName_test.go

```diff
@@ -46,7 +72,22 @@ func TestSaveUserName_ProcessBuiltinFunction(t *testing.T) {
 
 	_, err = coa.ProcessBuiltinFunction(nil, acc, vmInput)
 	require.Nil(t, err)
+	require.Equal(t, acc.GetUserName(), vmInput.Arguments[0])
 
 	_, err = coa.ProcessBuiltinFunction(nil, acc, vmInput)
 	require.Equal(t, ErrUserNameChangeIsDisabled, err)
+
+	coa.isChangeEnabled = func() bool {
+		return true
+	}
+
+	_, err = coa.ProcessBuiltinFunction(nil, acc, vmInput)
+	require.Equal(t, err, ErrCallerIsNotTheDNSAddress)
+
+	dnsAddrV2 := []byte("dnsV2")
+	coa.mapDnsV2Addresses[string(dnsAddrV2)] = struct{}{}
+	vmInput.CallerAddr = dnsAddrV2
+	_, err = coa.ProcessBuiltinFunction(nil, acc, vmInput)
+	require.Nil(t, err)
+	require.Equal(t, acc.GetUserName(), vmInput.Arguments[0])
```

Comment:
[optional]

Could've tried with a different userName (not `afafafaf...`) to check if it was changed

Reply:
done